module/sockets/grant-effect.mjs

```diff
@@ -0,0 +1,68 @@
+/**
+ * @typedef {object} GrantEffectConfig
+ * @property {string} event           The event name.
+ * @property {string} actor           The uuid of the actor to receive the effect.
+ * @property {object} effectData      Data used to create the effect.
+ * @property {string[]} userIds       The ids of the users that should perform the operation.
```

Comment:
It looks like `userIds` is required to be part of every socket event data definition, so that should be split off into a separate typedef that this one extends.